test/test_nn.py

```diff
@@ -13191,7 +13191,8 @@ def _test_GroupNorm_general(self, device, dtype=torch.float):
                 gn = nn.GroupNorm(g, shape[1])
 
     def _test_GroupNorm_cuda_half(self):
-        input = torch.zeros(2, 4, 3, 2, requires_grad=True).cuda().half().random_(1, 10)
```

Comment:
Example of an incorrect use of `random_` detected by this PR. I am not 100% sure whether the previous code has  performance implications (e.g. saving unnecessary tensors or doing extra work in the `backward`), but it is clearly an anti-pattern. Initialisation of tensors should not propagate gradients.